src/test/scala/ai/datahunters/md/filter/NotEmptyExtensionFilterSpec.scala

```diff
@@ -0,0 +1,115 @@
+      MetadataTagsSchemaConfig.MetadataCol, EmbeddedMetadataSchemaConfig.DirectoryNamesCol, EmbeddedMetadataSchemaConfig.TagNamesCol) === outputFields)
+    assert(collectedOutput.length === 0)
+  }
+
+  it should "pass one row" in {
+    val filter = new NotEmptyExtensionFilter(Seq[String](FileType1, FileType3))
+    val rdd = sparkSession.sparkContext.parallelize(Data)
+    val df = sparkSession.createDataFrame(rdd, Schema)
+    val outputDF = filter.execute(df)
+    val collectedOutput = outputDF.collect()
+    val outputFields = outputDF.schema.fields.map(_.name)
+    assert( Array(BinaryInputSchemaConfig.IDCol, BinaryInputSchemaConfig.BasePathCol, EmbeddedMetadataSchemaConfig.FileTypeCol,
+      MetadataTagsSchemaConfig.MetadataCol, EmbeddedMetadataSchemaConfig.DirectoryNamesCol, EmbeddedMetadataSchemaConfig.TagNamesCol) === outputFields)
+    assert(collectedOutput.length === 1)
+    val row1 = outputDF.collect()(0)
+    val fileTypeField = row1.get(row1.fieldIndex(EmbeddedMetadataSchemaConfig.FileTypeCol))
+    assert(FileType1 === fileTypeField)
+  }
+
+  it should "pass all rows" in {
+    val filter = new NotEmptyExtensionFilter(Seq[String](FileType1, FileType2))
+    val rdd = sparkSession.sparkContext.parallelize(Data)
+    val df = sparkSession.createDataFrame(rdd, Schema)
+    val outputDF = filter.execute(df)
+    val collectedOutput = outputDF.collect()
+    val outputFields = outputDF.schema.fields.map(_.name)
+    assert( Array(BinaryInputSchemaConfig.IDCol, BinaryInputSchemaConfig.BasePathCol, EmbeddedMetadataSchemaConfig.FileTypeCol,
+      MetadataTagsSchemaConfig.MetadataCol, EmbeddedMetadataSchemaConfig.DirectoryNamesCol, EmbeddedMetadataSchemaConfig.TagNamesCol) === outputFields)
+    assert(collectedOutput.length === 2)
+    val row1 = outputDF.collect()(0)
+    val row2 = outputDF.collect()(1)
+    val fileTypeField1 = row1.get(row1.fieldIndex(EmbeddedMetadataSchemaConfig.FileTypeCol))
+    val fileTypeField2 = row2.get(row2.fieldIndex(EmbeddedMetadataSchemaConfig.FileTypeCol))
+    assert(FileType1 === fileTypeField1)
+    assert(FileType2 === fileTypeField2)
+  }
+
+  it should "pass no rows" in {
+    val filter = new NotEmptyExtensionFilter(Seq[String](FileType3))
+    val rdd = sparkSession.sparkContext.parallelize(Data)
+    val df = sparkSession.createDataFrame(rdd, Schema)
+    val outputDF = filter.execute(df)
+    val collectedOutput = outputDF.collect()
+    val outputFields = outputDF.schema.fields.map(_.name)
+    assert( Array(BinaryInputSchemaConfig.IDCol, BinaryInputSchemaConfig.BasePathCol, EmbeddedMetadataSchemaConfig.FileTypeCol,
+      MetadataTagsSchemaConfig.MetadataCol, EmbeddedMetadataSchemaConfig.DirectoryNamesCol, EmbeddedMetadataSchemaConfig.TagNamesCol) === outputFields)
+    assert(collectedOutput.length === 0)
+  }
+
+}
+
+object NotEmptyExtensionFilterSpec {
+
+  val Id1 = "id1"
```

Comment:
@WojtekMin Looking at those sample strings growing in our project... I think we should add some common place to store such values. ScalaTest provides something for it but let's leave it now.